lewis/adapters/stream.py

```diff
@@ -48,20 +48,13 @@ def found_terminator(self):
         self.buffer = []
 
         try:
-            match = None
-            for cmd in self.target.commands:
-                match = cmd.pattern.match(request)
-                if match:
-                    groups = match.groups()
-                    func = getattr(self.target, cmd.method)
-
-                    args = cmd.map_arguments(groups)
-                    reply = cmd.return_mapping(func(*args))
-                    break
-
-            if match is None:
+            cmd = next((cmd for cmd in self.target.commands if cmd.can_process(request)), None)
+
+            if cmd is None:
```

Comment:
Now this is much better from my point of view. Get's the first matching `Cmd` or `None` if none is found. In that case, as before, raise `RuntimeError` so it can be handled (or not). Otherwise continue processing request.